caom2utils/caom2utils/fits2caom2.py

```diff
@@ -3035,6 +3226,34 @@ def get_cadc_headers(uri, subject=None):
     return headers
 
 
+def get_vos_headers(uri, subject=None):
+    """
+    Creates the FITS headers object from a vospace file. 
+    The function uses cutouts to retrieve the miniumum amount of data,
+     minimizing the transfer time.
+    :param uri: vos URI
+    :param subject: user credentials. Anonymous if subject is None
+    :return: List of headers corresponding to each extension. Each header is
+    of astropy.wcs.Header type - essentially a dictionary of FITS keywords.
+    """
+    if uri.startswith('vos'):
+        if subject is not None and subject.certificate is not None:
+            client = Client(subject.certificate)
+        else:
+            client = Client()
+
+        # make the smallest cutout possible, to get the least amount of data
+        # transferred, then transfer it to a temporary file
+        #
+        uri_with_cutout = '{}[1:1,1:1]'.format(uri)
+        temp_filename = tempfile.NamedTemporaryFile()
+        client.copy(uri_with_cutout, temp_filename.name)
+        return _get_headers_from_fits(temp_filename.name)
+    else:
+        # this should be a programming error by now
+        raise NotImplementedError('Only ad type URIs supported')
```

Comment:
Typo: ad->vos

Reply:
Fixed. Also line 3299, according to the modified version of the code.